streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -606,18 +606,29 @@ abstract class ZStream[-R, +E, +O](val process: ZManaged[R, Nothing, ZIO[R, Opti
    * whether two elements are equal.
    */
   def changesWith(f: (O, O) => Boolean): ZStream[R, E, O] =
+    changesWithZIO((o, o1) => ZIO.succeedNow(f(o, o1)))
```

Comment:
This specialized variant may be higher enough performance that it's worth keeping around. /cc @iravid

Reply:
Yeah, tend to agree. That's also important in the channels version where avoiding ZIO effects skips ZIO entirely.